src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -580,7 +580,7 @@ protected virtual string[] GetTypeProperties(Type type)
             var itemType = jsonSchemaAttribute?.ArrayItem ?? contextualType.OriginalType.GetEnumerableItemType();
             if (itemType != null)
             {
-                var contextualItemType = itemType.ToContextualType();
+                var contextualItemType = contextualType.ElementType ?? itemType.ToContextualType();
```

Comment:
not sure why `contextualType.ElementType` is `null` for some other cases. Maybe there's something that should be fixed in Namotion.Reflection ?

For example, this doesn't work with `List<string?>`

Reply:
Here we need to use the new EnumerableItemType, right?